### PR TITLE
OPENEUROPA-1974: Update ecl to 2.11.0

### PR DESCRIPTION
**Pull request by voidtek**
### Description

Update ecl to 2.11.0.

### CODE REVIEW

File: oe_theme.libraries.yml [hunk 1]
```diff
@@ -36,18 +25,6 @@ context_nav:
   dependencies:
     - core/drupal
     - oe_theme/base
-file_translations:
-  js:
-    js/file_translations.js: {}
-  dependencies:
-    - core/drupal
-    - oe_theme/base
-inpage_navigation:
```
Comment (imanoleguskiza) [inline oe_theme.libraries.yml]: The library was removed but its still being loaded in paragraph--oe-content-row--variant-inpage_navigation.html.twig

File: oe_theme.libraries.yml [hunk 2]
```diff
@@ -36,18 +25,6 @@ context_nav:
...
-  js:
-    js/inpage_navigation.js: {}
-  dependencies:
-    - core/drupal
-    - oe_theme/base
 lang_select_pages:
```
Comment (imanoleguskiza) [inline oe_theme.libraries.yml]: The js was removed but the library is still here.

Comment (imanoleguskiza) [reply oe_theme.libraries.yml]: Also its still used on links--oe-multilingual-content-language-block.html.twig

File: oe_theme.libraries.yml [hunk 3]
```diff
@@ -1,23 +1,12 @@
 base:
   js:
     dist/scripts/ecl-ec-preset-website.js: {}
+    js/ecl_auto_init.js: {}
 style:
   css:
     theme:
       dist/styles/ecl-ec-preset-website.css: {}
       css/style.css: {}
-accordions:
```
Comment (imanoleguskiza) [inline oe_theme.libraries.yml]: The library was removed but its still being loaded on the accordion.ui_patterns.yml file

File: oe_theme.libraries.yml [hunk 4]
```diff
@@ -1,23 +1,12 @@
...
-  js:
-    js/accordions.js: {}
-  dependencies:
-    - core/drupal
-    - oe_theme/base
-message:
```
Comment (imanoleguskiza) [inline oe_theme.libraries.yml]: Library was removed but its being loaded on status-messages.html.twig

File: oe_theme.libraries.yml [hunk 5]
```diff
@@ -36,18 +25,6 @@ context_nav:
   dependencies:
     - core/drupal
     - oe_theme/base
-file_translations:
```
Comment (imanoleguskiza) [inline oe_theme.libraries.yml]: Library was removed but is still being loaded on file_translation.ui_patterns.yml

File: oe_theme.libraries.yml [hunk 6]
```diff
@@ -66,9 +43,3 @@ tables:
   dependencies:
     - core/drupal
     - oe_theme/base
-timelines:
```
Comment (imanoleguskiza) [inline oe_theme.libraries.yml]: Library was remved but is still being loaded on timeline.ui_patterns.yml

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]
I noticed some of the JS libraries where kept, any reason for that?

File: oe_theme.libraries.yml (hunk 1 above)
Comment (voidtek) [reply oe_theme.libraries.yml]: Very good catch.

File: oe_theme.libraries.yml (hunk 2 above)
Comment (voidtek) [reply oe_theme.libraries.yml]: Very good catch.

File: oe_theme.libraries.yml (hunk 3 above)
Comment (voidtek) [reply oe_theme.libraries.yml]: Very good catch.

File: oe_theme.libraries.yml (hunk 4 above)
Comment (voidtek) [reply oe_theme.libraries.yml]: Very good catch.

File: oe_theme.libraries.yml (hunk 5 above)
Comment (voidtek) [reply oe_theme.libraries.yml]: Very good catch.

File: oe_theme.libraries.yml (hunk 6 above)
Comment (voidtek) [reply oe_theme.libraries.yml]: Very good catch.

**Review by imanoleguskiza — DISMISSED** [review]

File: oe_theme.theme [hunk 7]
```diff
@@ -646,11 +646,13 @@ function oe_theme_preprocess_pattern_accordion(array &$variables): void {
...
-  if (!empty($variables['video']['sources'])) {
-    // We need to rename the variables that the Patterns module uses.
-    foreach ($variables['video']['sources'] as &$source) {
-      $source['type'] = $source['src_type'];
-    }
+  if (empty($variables['video']['sources'])) {
+    return;
+  }
+
+  // We need to rename the variables that the Patterns module uses.
```
Comment (ademarco) [inline oe_theme.theme]: Maybe this is more clear?
>    // Map UI Patterns fields to ECL component fields.

File: patches/@ecl-twig/ec-component-accordion2+2.8.0.patch [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+--- a/node_modules/@ecl-twig/ec-component-accordion2/accordion2.html.twig
++++ b/node_modules/@ecl-twig/ec-component-accordion2/accordion2.html.twig
+@@ -47,6 +47,7 @@
+ <div
+   class="{{ _css_class }}"{{ _extra_attributes|raw }}
+   data-ecl-accordion2
++  data-ecl-auto-init="Accordion2"
+ >
+   {% if _items is not empty %}
+     {% for _item in _items %}
```
Comment (ademarco) [inline patches/@ecl-twig/ec-component-accordion2+2.8.0.patch]: @voidtek we should maybe open follow up issues on ECL Twig board to review and, possibly, include these patches in their project.

File: templates/form/form-element-label.html.twig [hunk 9]
```diff
@@ -10,11 +10,11 @@
...
     title_display == 'after' ? 'option',
-    title_display == 'invisible' ? 'visually-hidden ecl-form-label--hidden',
+    title_display == 'invisible' ? 'visually-hidden',
     required ? 'js-form-required',
     required ? 'form-required',
   ]
 %}
 {% if title is not empty or required -%}
-  <label{{ attributes.addClass(classes) }}>{{ title }}</label>
+  <label{{ attributes.addClass(classes) }}{{ title_display == 'invisible' ? ' hidden ' }}>{{ title }}</label>
```
Comment (ademarco) [inline templates/form/form-element-label.html.twig]: Not needed space after `hidden ': 
image

**Review by ademarco — CHANGES_REQUESTED** [review]

File: templates/form/radios.html.twig [hunk 10]
```diff
@@ -18,8 +18,9 @@
 <div{{ attributes.setAttribute("role", "radiogroup").addClass(classes) }}>
   {%- if _label is not empty %}
     <div
-      class="ecl-form-label{{ hide_label ? ' ecl-form-label--hidden'}}"
+      class="ecl-form-label"
       id="{{ _label_id }}"
+      {{ hide_label ? 'hidden'}}
```
Comment (ademarco) [inline templates/form/radios.html.twig]: If not set this will produce an empty line in the markup, maybe we put this one on the line above?

**Review by ademarco — CHANGES_REQUESTED** [review]

File: oe_theme.theme (hunk 7 above)
Comment (voidtek) [reply oe_theme.theme]: done.

File: patches/@ecl-twig/ec-component-accordion2+2.8.0.patch (hunk 8 above)
Comment (voidtek) [reply patches/@ecl-twig/ec-component-accordion2+2.8.0.patch]: Already created, it's the ticket to update ecl-twig to 2.11.

File: templates/form/form-element-label.html.twig (hunk 9 above)
Comment (voidtek) [reply templates/form/form-element-label.html.twig]: done.

File: templates/form/radios.html.twig (hunk 10 above)
Comment (voidtek) [reply templates/form/radios.html.twig]: done

File: patches/@ecl-twig/ec-component-accordion2+2.8.0.patch (hunk 8 above)
Comment (ademarco) [reply patches/@ecl-twig/ec-component-accordion2+2.8.0.patch]: Ah nice!

**Review by ademarco — DISMISSED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**